### PR TITLE
feat: create `Context.has_permissions()` & `Member.has_permissions()`

### PR DESCRIPTION
**Pull request by mAxYoLo01**
## About

Creation of `Context.has_permissions()` & `Member.has_permissions()`.
Also, allowed `Member.get_guild_permissions()` to not pass in the guild_id thanks to #1025 .

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues) (If existent):.
  - resolves #
- I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/member.py [hunk 1]
```diff
@@ -521,6 +523,22 @@ async def get_guild_permissions(self, guild: "Guild") -> Permissions:
...
+            guild = guild_id
+
+        else:
+            if guild_id is MISSING:
+                _guild_id = self.guild_id
+                if isinstance(_guild_id, LibraryException):
+                    raise _guild_id
+
+            else:
+                _guild_id = int(guild_id)
```
Comment (sillyElvy) [inline interactions/api/models/member.py]:
```suggestion
        if guild_id is MISSING:
            _guild_id = self.guild_id
            if isinstance(_guild_id, LibraryException):
                raise _guild_id

        else:
            _guild_id = int(guild_id) if not isinstance(guild_id, Guild) else int(guild_id.id)
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/channel.py [hunk 2]
```diff
@@ -1231,13 +1231,9 @@ async def get_permissions_for(self, member: "Member") -> Permissions:
         if not self.guild_id:
             return Permissions.DEFAULT
 
-        from .guild import Guild
+        permissions = await member.get_guild_permissions(self.guild_id)
```
Comment (sillyElvy) [inline interactions/api/models/channel.py:1234]: Unsure but I think channel has guild id only in certain events, have you checked if this exists when you get the channel manually?

Comment (mAxYoLo01) [reply interactions/api/models/channel.py:1234]: If `guild_id` is None, it returns default permissions (lines above). Unsure if that's what you meant?
But yes I have tried it by getting the channel and member, and running `channel.get_permissions_for(member)`

Comment (sillyElvy) [reply interactions/api/models/channel.py:1234]: Oh I didn't see that, Nevermind then

**Review by sillyElvy — APPROVED** [review]
I think it's fine

**Pull request merged**